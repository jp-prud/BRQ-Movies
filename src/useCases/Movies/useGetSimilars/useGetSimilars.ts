import { MoviesService } from "@services"
import { useQuery } from "@tanstack/react-query"
import { MovieProps, QueryKeys } from "@types"

export function useGetSimilars(movieId: MovieProps['id']) {
  const { getSimilars } = MoviesService()

  const { data: similars, isLoading, isError } = useQuery({
    queryKey: [QueryKeys.listSimilars, movieId],
    queryFn: () => getSimilars(movieId)
  })

  return {
    similars, isLoading, isError
  }
}