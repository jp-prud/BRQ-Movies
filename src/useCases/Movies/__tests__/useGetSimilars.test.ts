import { renderHook, waitFor } from '@tests';

import { useGetSimilars } from '../useGetSimilars/useGetSimilars';

describe('useGetSimilars', () => {
  it('should get similar movies by id', async () => { 
    const { result } = renderHook(() => useGetSimilars(533535));

    await waitFor(() =>
      expect(result.current.isLoading).toBe(false),
    );

    expect(result.current.similars).toMatchSnapshot();
    expect(result.current.isError).toBe(false);
  });
});
