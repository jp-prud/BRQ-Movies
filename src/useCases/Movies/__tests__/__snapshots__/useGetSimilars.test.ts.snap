// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useGetSimilars should get similar movies by id 1`] = `
{
  "data": [
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/rlvM2us5esS6TXB581vZx6BzHzO.jpg",
      "genreIds": [
        28,
        12,
      ],
      "genres": undefined,
      "id": 401461,
      "originalLanguage": "en",
      "originalTitle": "The Phantom",
      "overview": "Two expeditions are trying to reach the Lost City of Zoloz -- one headed by Professor Davidson, a scientist who wants to establish an archaeological site, and the other by a greedy treasure hunter who wants to keep the fabled treasures of the city for himself. An agent of a foreign power also wants to establish a secret airbase there, so he stirs up the natives against The Phantom, who has been able to get them to stay peaceful so far. When The Phantom is murdered, his son takes his place and sets out to restore peace to the jungle and stop the agents' and the treasure hunters' nefarious plans.",
      "popularity": 4.001,
      "posterPath": "https://image.tmdb.org/t/p/w185/uSZrqoprWkiK770H7XUjLXzyGzD.jpg",
      "releaseDate": "1943-12-24",
      "title": "The Phantom",
      "video": false,
      "voteAverage": 6.222,
      "voteCount": 9,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/2ex2beZ4ssMeOduLD0ILzXKCiep.jpg",
      "genreIds": [
        28,
        12,
        878,
        14,
      ],
      "genres": undefined,
      "id": 246655,
      "originalLanguage": "en",
      "originalTitle": "X-Men: Apocalypse",
      "overview": "After the re-emergence of the world's first mutant, world-destroyer Apocalypse, the X-Men must unite to defeat his extinction level plan.",
      "popularity": 143.369,
      "posterPath": "https://image.tmdb.org/t/p/w185/2mtQwJKVKQrZgTz49Dizb25eOQQ.jpg",
      "releaseDate": "2016-05-18",
      "title": "X-Men: Apocalypse",
      "video": false,
      "voteAverage": 6.516,
      "voteCount": 12723,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/lcWXnH968CpOOzhqsFIpKn1z9Ft.jpg",
      "genreIds": [
        878,
        28,
      ],
      "genres": undefined,
      "id": 11795,
      "originalLanguage": "en",
      "originalTitle": "Guyver: Dark Hero",
      "overview": "Sean Barker became the unwilling host to an alien bio-armor known as the Guyver. A year ago he destroyed the Kronos Corporation, an organization of mutants who want the Guyver. Now he is trying to find why the Guyver unit forces him to fight and kill evil.",
      "popularity": 9.6,
      "posterPath": "https://image.tmdb.org/t/p/w185/bHbyCAJs9w6znVhgdyRKHMcbGQ4.jpg",
      "releaseDate": "1994-04-20",
      "title": "Guyver: Dark Hero",
      "video": false,
      "voteAverage": 5.6,
      "voteCount": 76,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/spzbYAXGxAGUoBekx3bHsbKzVQu.jpg",
      "genreIds": [
        35,
        28,
        878,
      ],
      "genres": undefined,
      "id": 11918,
      "originalLanguage": "en",
      "originalTitle": "Superhero Movie",
      "overview": "Rick Riker is a nerdy teen imbued with superpowers by a radioactive dragonfly. And because every hero needs a nemesis, enter Lou Landers, aka the villainously goofy Hourglass.",
      "popularity": 26.631,
      "posterPath": "https://image.tmdb.org/t/p/w185/4MtvQdh5VP7jMSydhggp1P80FMj.jpg",
      "releaseDate": "2008-03-28",
      "title": "Superhero Movie",
      "video": false,
      "voteAverage": 5.249,
      "voteCount": 1764,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/AoFo9HeJU7jiHAUNawpwhs3zQbd.jpg",
      "genreIds": [
        37,
        28,
        12,
      ],
      "genres": undefined,
      "id": 11967,
      "originalLanguage": "en",
      "originalTitle": "Young Guns",
      "overview": "A group of young gunmen, led by Billy the Kid, become deputies to avenge the murder of the rancher who became their benefactor. But when Billy takes their authority too far, they become the hunted.",
      "popularity": 36.387,
      "posterPath": "https://image.tmdb.org/t/p/w185/stvh1A3mj8NYyczE7RvwdtDRCo2.jpg",
      "releaseDate": "1988-02-17",
      "title": "Young Guns",
      "video": false,
      "voteAverage": 6.7,
      "voteCount": 881,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/vc4GTzgFVPfDWkkbBdjYLpcbrUy.jpg",
      "genreIds": [
        35,
        10751,
        14,
      ],
      "genres": undefined,
      "id": 1203336,
      "originalLanguage": "nl",
      "originalTitle": "Superkrachten voor je hoofd",
      "overview": "A boy uncertain due to a disability who wants to meet his big superhero in the hope that he can make him walk again. He goes on an adventure with his grandma, much to the reluctance of his overprotective parents.",
      "popularity": 4.572,
      "posterPath": "https://image.tmdb.org/t/p/w185/4H421EPnAhdtt7JV3eFfwB6gh0J.jpg",
      "releaseDate": "2024-07-10",
      "title": "Superkrachten voor je hoofd",
      "video": false,
      "voteAverage": 0,
      "voteCount": 0,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/lZCDvCaCrnFK8EcX5TTmk72PzCy.jpg",
      "genreIds": [
        28,
        35,
        14,
      ],
      "genres": undefined,
      "id": 11817,
      "originalLanguage": "en",
      "originalTitle": "Bulletproof Monk",
      "overview": "A mysterious and immortal Tibetan kung fu master, who has spent the last 60 years traveling around the world protecting the ancient Scroll of the Ultimate, mentors a selfish street kid in the ancient intricacies of kung fu.",
      "popularity": 30.274,
      "posterPath": "https://image.tmdb.org/t/p/w185/kHWv16c4uDmiEujICSBdQ52TH55.jpg",
      "releaseDate": "2003-03-28",
      "title": "Bulletproof Monk",
      "video": false,
      "voteAverage": 5.516,
      "voteCount": 889,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/mWzgMPlQcFZ6lZ449673LS2aCHX.jpg",
      "genreIds": [
        10751,
        16,
        28,
        878,
      ],
      "genres": undefined,
      "id": 251768,
      "originalLanguage": "en",
      "originalTitle": "JLA Adventures: Trapped in Time",
      "overview": "Get ready for a battle of the ages when the Justice League faces off against its archenemies, the Legion of Doom, in an all-new movie from DC Comics. A mysterious being known as the Time Trapper arises, and a sinister plan led by Lex Luthor sends the Legion of Doom back in time to eliminate Superman, Wonder Woman and Batman before they become super heroes. For Aquaman, Flash and Cyborg, along with teen super heroes Karate Kid and Dawnstar, the stakes have never been higher, the rescue mission never deadlier. So join the fight for the future as the Justice League confronts its ultimate challenge… the threat of having never existed!",
      "popularity": 12.362,
      "posterPath": "https://image.tmdb.org/t/p/w185/qchNuTbnleCtlIB8wyV03f36VuF.jpg",
      "releaseDate": "2014-01-21",
      "title": "JLA Adventures: Trapped in Time",
      "video": false,
      "voteAverage": 6.6,
      "voteCount": 213,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185null",
      "genreIds": [
        28,
        12,
      ],
      "genres": undefined,
      "id": 251500,
      "originalLanguage": "ja",
      "originalTitle": "月光仮面",
      "overview": "The Moon Mask Rider is a tokusatsu movie produced by Purumie International/Herald Enterprises and distributed by Nippon Herald Pictures, was released theatrically on March 14, 1981. Considered Japan's answer to the American box-office fiasco, The Legend of the Lone Ranger (released the same year), this updated version of the Moonlight Mask legend bombed at the Japanese box-office. Daisuke Kuwahara (who, like Klinton Spilsbury , disappeared from doing films) plays George Owara (Moon Mask Rider's new alter-ego), and the rest of the cast made up of veteran action starlets: Sue Shihomi, Daijiro Harada and Takayuki Godai.",
      "popularity": 3.202,
      "posterPath": "https://image.tmdb.org/t/p/w185/zKXs8v6HhAwKoESm2slWYMmrzC3.jpg",
      "releaseDate": "1981-03-14",
      "title": "The Moon Mask Rider",
      "video": false,
      "voteAverage": 5.2,
      "voteCount": 2,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/jYqOnJDPI7QD9cGbE6CwDTp1tID.jpg",
      "genreIds": [
        16,
        28,
        12,
      ],
      "genres": undefined,
      "id": 251519,
      "originalLanguage": "en",
      "originalTitle": "Son of Batman",
      "overview": "Batman learns he has a violent, unruly pre-teen son, secretly raised by the terrorist group known as The League of Assassins.",
      "popularity": 19.625,
      "posterPath": "https://image.tmdb.org/t/p/w185/yHx8OLSKc3VtIBB5lIe90c0fHOX.jpg",
      "releaseDate": "2014-05-13",
      "title": "Son of Batman",
      "video": false,
      "voteAverage": 7.046,
      "voteCount": 1031,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185null",
      "genreIds": [
        28,
      ],
      "genres": undefined,
      "id": 405561,
      "originalLanguage": "en",
      "originalTitle": "Volume",
      "overview": "Superhero movie about a man who can control soundwaves.",
      "popularity": 1.669,
      "posterPath": "https://image.tmdb.org/t/p/w185null",
      "releaseDate": "",
      "title": "Volume",
      "video": false,
      "voteAverage": 0,
      "voteCount": 0,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185null",
      "genreIds": [
        14,
        28,
      ],
      "genres": undefined,
      "id": 564094,
      "originalLanguage": "bn",
      "originalTitle": "KEN The Super Hero",
      "overview": "Ajooba's main magician agnila creat different creature to destroy ampara. ken is a powerful superhero of the planet ampara. He has lots of magical power. in this story ken saved the world from ajooba.",
      "popularity": 1.037,
      "posterPath": "https://image.tmdb.org/t/p/w185/4XsCfkWSR7UuOCcbiswCMrIBKgy.jpg",
      "releaseDate": "2018-11-23",
      "title": "KEN The Super Hero",
      "video": false,
      "voteAverage": 0,
      "voteCount": 0,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/i5hp04Eoou5YJWsJsmL7Zsw7Yfl.jpg",
      "genreIds": [
        28,
        14,
        53,
      ],
      "genres": undefined,
      "id": 10546,
      "originalLanguage": "en",
      "originalTitle": "The Crow: City of Angels",
      "overview": "A murder victim is brought back to life by a mysterious crow. With the help of a beautiful woman named Sarah, he exacts revenge on his killers -- only to realize his enemy has discovered the one weakness that can destroy him forever.",
      "popularity": 17.662,
      "posterPath": "https://image.tmdb.org/t/p/w185/wcWXBujfOUKYbWRnfnAUJaqxRrn.jpg",
      "releaseDate": "1996-08-29",
      "title": "The Crow: City of Angels",
      "video": false,
      "voteAverage": 5.6,
      "voteCount": 510,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/sGDXh6QU6CRqHVXfoYOBwqEnBVO.jpg",
      "genreIds": [
        28,
        36,
        10752,
      ],
      "genres": undefined,
      "id": 10590,
      "originalLanguage": "en",
      "originalTitle": "We Were Soldiers",
      "overview": "The story of the first major battle of the American phase of the Vietnam War and the soldiers on both sides that fought it.",
      "popularity": 40.054,
      "posterPath": "https://image.tmdb.org/t/p/w185/q2LeLzw4q9HTelkKx29dyTMf5zK.jpg",
      "releaseDate": "2002-03-01",
      "title": "We Were Soldiers",
      "video": false,
      "voteAverage": 7.143,
      "voteCount": 1992,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/6UPAuIIL3sjrvuRxsrWqG13G77R.jpg",
      "genreIds": [
        12,
        35,
        10751,
        10749,
      ],
      "genres": undefined,
      "id": 10603,
      "originalLanguage": "en",
      "originalTitle": "George of the Jungle",
      "overview": "Deep in the African jungle, a baby named George, the sole survivor of a plane crash, is raised by gorillas. George grows up to be a buff and lovable klutz who has a rain forest full of animal friends: Tookie, his big-beaked toucan messenger; Ape, a witty talking gorilla; and Shep, a peanut-loving pooch of an elephant. But when poachers mess with George's pals, the King Of Swing swings into action.",
      "popularity": 43.614,
      "posterPath": "https://image.tmdb.org/t/p/w185/lWp8hUqE4oLPxsYgilXoYoVThfU.jpg",
      "releaseDate": "1997-07-15",
      "title": "George of the Jungle",
      "video": false,
      "voteAverage": 5.6,
      "voteCount": 2032,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/shl0ELODK3VgHr7pu8VTz2a8olW.jpg",
      "genreIds": [
        28,
        18,
      ],
      "genres": undefined,
      "id": 10617,
      "originalLanguage": "cn",
      "originalTitle": "黃飛鴻",
      "overview": "Set in late 19th century Canton, this martial arts film depicts the stance taken by the legendary martial arts hero Wong Fei-Hung against foreign forces' plundering of China.",
      "popularity": 22.204,
      "posterPath": "https://image.tmdb.org/t/p/w185/dkBQC0jmkmTOJJMgwsBdgkzZ6Ry.jpg",
      "releaseDate": "1991-08-15",
      "title": "Once Upon a Time in China",
      "video": false,
      "voteAverage": 7.136,
      "voteCount": 405,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/rNux3fw0jC9wOi3oIxJH2X1Sbap.jpg",
      "genreIds": [
        37,
        35,
      ],
      "genres": undefined,
      "id": 11072,
      "originalLanguage": "en",
      "originalTitle": "Blazing Saddles",
      "overview": "A town—where everyone seems to be named Johnson—stands in the way of the railroad. In order to grab their land, robber baron Hedley Lamarr sends his henchmen to make life in the town unbearable. After the sheriff is killed, the town demands a new sheriff from the Governor, so Hedley convinces him to send the town the first black sheriff in the west.",
      "popularity": 37.499,
      "posterPath": "https://image.tmdb.org/t/p/w185/vNw1gOEDdYTDeNMuuq8OmiEHrfH.jpg",
      "releaseDate": "1974-02-07",
      "title": "Blazing Saddles",
      "video": false,
      "voteAverage": 7.256,
      "voteCount": 1808,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/dfxPajRKhApSMfv31Sq14loL8Ck.jpg",
      "genreIds": [
        18,
        28,
        53,
      ],
      "genres": undefined,
      "id": 11128,
      "originalLanguage": "en",
      "originalTitle": "Ladder 49",
      "overview": "Under the watchful eye of his mentor, Captain Mike Kennedy, probationary firefighter Jack Morrison matures into a seasoned veteran at a Baltimore fire station. However, Jack has reached a crossroads  as the sacrifices he's made have put him in harm's way innumerable times and significantly impacted his relationship with his wife and kids.",
      "popularity": 23.41,
      "posterPath": "https://image.tmdb.org/t/p/w185/aJWQNUUHVcI7kvlVdUzPS9J8XHt.jpg",
      "releaseDate": "2004-10-01",
      "title": "Ladder 49",
      "video": false,
      "voteAverage": 6.392,
      "voteCount": 752,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185/xFKBurykrSIkIC0vmlaZ7VDY6HT.jpg",
      "genreIds": [
        12,
        35,
        10751,
      ],
      "genres": undefined,
      "id": 11459,
      "originalLanguage": "en",
      "originalTitle": "Sky High",
      "overview": "Set in a world where superheroes are commonly known and accepted, young Will Stronghold, the son of the Commander and Jetstream, tries to find a balance between being a normal teenager and an extraordinary being.",
      "popularity": 49.961,
      "posterPath": "https://image.tmdb.org/t/p/w185/yk51E2OxA9zUdRqK0YvPTx6lfa7.jpg",
      "releaseDate": "2005-07-29",
      "title": "Sky High",
      "video": false,
      "voteAverage": 6.241,
      "voteCount": 2248,
    },
    {
      "adult": false,
      "backdropPath": "https://image.tmdb.org/t/p/w185null",
      "genreIds": [
        878,
        10770,
      ],
      "genres": undefined,
      "id": 562429,
      "originalLanguage": "en",
      "originalTitle": "The Power Within",
      "overview": "A prospective about a daredevil flyer, barnstorming for county fairs, who becomes a human dynamo after being struck by lightning and is menaced by enemy agents determined to find out the secret of his incredible powers.",
      "popularity": 2.989,
      "posterPath": "https://image.tmdb.org/t/p/w185/yw4YoyxIAvJFU4zc0AqrotBbqBF.jpg",
      "releaseDate": "1979-05-11",
      "title": "The Power Within",
      "video": false,
      "voteAverage": 0,
      "voteCount": 0,
    },
  ],
  "meta": {
    "currentPage": 1,
    "firstPage": 1,
    "hasNextPage": true,
    "hasPreviousPage": false,
    "lastPage": 9464,
    "total": 189270,
  },
}
`;
